### PR TITLE
added eslint rules, removed warnings

### PR DESCRIPTION
**Pull request by IAmMorrow**
added react-hooks eslint rule. fixed newly generated warnings (the codebase was mostly compliant already)

### Type

Code Quality Improvement

### Context

LL-1639

### Parts of the app affected / Test plan

eslint configuration, the whole app coding style

### CODE REVIEW

File: src/components/modals/AddAccounts/steps/03-step-import.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ const SectionAccounts = ({ defaultSelected, ...rest }: *) => {
     if (defaultSelected && rest.onSelectAll) {
       rest.onSelectAll(rest.accounts)
     }
-  }, [])
+  }, [defaultSelected, rest])
```
Comment (gre) [inline src/components/modals/AddAccounts/steps/03-step-import.js]: this one will be problematic I think. rest always changes, right? this is meant to be exec once on mount.

Comment (IAmMorrow) [reply src/components/modals/AddAccounts/steps/03-step-import.js]: @gre hmmm but if we actually use `onSelectAll` and `accounts` in that component should we not declare them as proper props ? I think `rest` should be passed "as is" and not be interacted with

Comment (IAmMorrow) [reply src/components/modals/AddAccounts/steps/03-step-import.js]: This is OK anyway, as non providing an input arg solve the problem. The eslint hook rule doesn't like empty arrays

**Comment by gre** [comment]
@Arnaud97234 you can check all changed component still work fine. @IAmMorrow can help if need more context.

**Pull request merged**